core/channel-upgrade.go

```diff
@@ -133,55 +133,48 @@ func ExecuteChannelUpgrade(ctx context.Context, pathName string, src, dst *Prova
 	logger := GetChannelPairLogger(src, dst)
 	defer logger.TimeTrack(time.Now(), "ExecuteChannelUpgrade")
 
-	ticker := time.NewTicker(interval)
-	defer ticker.Stop()
-
 	failures := 0
 	firstCall := true
-	for {
+	return runUntilComplete(ctx, interval, func() (bool, error) {
```

Comment:
I also thought of using `retry.Do` like https://github.com/abicky/yui-relayer/commit/d7b73e0d6f3c6dae891052fcf6143c94222e7909, but I felt a little weird about [returning an error when proceeding to the next step](https://github.com/abicky/yui-relayer/commit/d7b73e0d6f3c6dae891052fcf6143c94222e7909#diff-ef4ec23f51199931750de82658cf40eb63726ceeafea6a8b3f5d5ead14b97343R167)...